### PR TITLE
Implemented dark mode

### PR DESCRIPTION
**Pull request by 3milyfz**
- Added dark mode toggle
- Colors can be changed by the team!

Sample dashboard:

image

image

### CODE REVIEW

**Review by GregoryGismondi — APPROVED** [review]
# Reviewing Big Changes From Testing
### Package.json Changes
- MUI version being updated did not appear to have any effects, mostly because the version changes are mainly targeting bug fixes and Joy UI (which we are not using for this project). There may be some bugs that we were relying on in our code, but if no issues arose from testing then it should be fine.  

### src/App.js
- Removing UseState import did not have an impact as its import and usage was on a previous version of App.js that has long been changed
- Wrapping routing code in DarkModeProvider did not have any conflicts with the screens as the code just adds in an extra variable to tell the screens what mode they are on (light or dark)

### src/components/NavBar
- Navigation bar layout still looks good, but may need to be updated when other pull requests are completed

### All Screens
- Found a bug at the bottom of the PDF not displaying the correct colour, but a fix to the bug was tested and committed

# Going Forward
- Future screens will need to follow similar structure to the current screen as there are more moving parts
- Will need to edit other current pull requests so that they fit the new version of main

Overall, the dark mode changes look beautiful. Thank you so much!
tl;dr: LGTM
